java-checks/src/main/java/org/sonar/java/checks/ReturnEmptyArrayNotNullCheck.java

```diff
@@ -46,49 +44,10 @@
 @Rule(key = "S1168")
 public class ReturnEmptyArrayNotNullCheck extends IssuableSubscriptionVisitor {
 
-  private static final Set<String> COLLECTION_TYPES = SetUtils.immutableSetOf(
-    "Collection",
-    "BeanContext",
-    "BeanContextServices",
-    "BlockingDeque",
-    "BlockingQueue",
-    "Deque",
-    "List",
-    "NavigableSet",
-    "Queue",
-    "Set",
-    "SortedSet",
-    "AbstractCollection",
-    "AbstractList",
-    "AbstractQueue",
-    "AbstractSequentialList",
-    "AbstractSet",
-    "ArrayBlockingQueue",
-    "ArrayDeque",
-    "ArrayList",
-    "AttributeList",
-    "BeanContextServicesSupport",
-    "BeanContextSupport",
-    "ConcurrentLinkedQueue",
-    "ConcurrentSkipListSet",
-    "CopyOnWriteArrayList",
-    "CopyOnWriteArraySet",
-    "DelayQueue",
-    "EnumSet",
-    "HashSet",
-    "JobStateReasons",
-    "LinkedBlockingDeque",
-    "LinkedBlockingQueue",
-    "LinkedHashSet",
-    "LinkedList",
-    "PriorityBlockingQueue",
-    "PriorityQueue",
-    "RoleList",
-    "RoleUnresolvedList",
-    "Stack",
-    "SynchronousQueue",
-    "TreeSet",
-    "Vector");
+  private static final List<String> COLLECTION_TYPES = Arrays.asList(
+    "java.util.Collection",
+    "javax.print.attribute.standard.JobStateReason"
```

Comment:
Remove `JobStateReason`, it is not a collection. `JobStateReasons` is, but it also inherits from Collection.
So in the end, we can simplify to only support "java.util.Collection"